packages/@lwc/shared/src/fields.ts

```diff
@@ -6,35 +6,41 @@
  */
 import { create, isUndefined } from './language';
 
-/**
+/*
+ * Typescript doesn't allow indexing using Symbol, aliasing the Field type to any for now.
+ * Details: https://github.com/microsoft/TypeScript/issues/1863
+ */
+type HiddenField = any;
+
+/*
  * In IE11, symbols are expensive.
  * Due to the nature of the symbol polyfill. This method abstract the
  * creation of symbols, so we can fallback to string when native symbols
  * are not supported. Note that we can't use typeof since it will fail when transpiling.
  */
 const hasNativeSymbolsSupport = Symbol('x').toString() === 'Symbol(x)';
 
-export function createFieldName(key: string, namespace: string): symbol {
-    // @ts-ignore: using a string as a symbol for perf reasons
+export function createFieldName(key: string, namespace: string): HiddenField {
     return hasNativeSymbolsSupport ? Symbol(key) : `$$lwc-${namespace}-${key}$$`;
 }
 
-const hiddenFieldsMap: WeakMap<any, Record<symbol | string, any>> = new WeakMap();
+const hiddenFieldsMap: WeakMap<any, Record<HiddenField, any>> = new WeakMap();
 
-export function setHiddenField(o: any, fieldName: symbol, value: any) {
+export function setHiddenField(o: any, field: HiddenField, value: any): void {
     let valuesByField = hiddenFieldsMap.get(o);
+
     if (isUndefined(valuesByField)) {
-        valuesByField = create(null) as Record<symbol, any>;
-        hiddenFieldsMap.set(o, valuesByField);
+        valuesByField = create(null);
+        hiddenFieldsMap.set(o, valuesByField!);
     }
-    // @ts-ignore https://github.com/microsoft/TypeScript/issues/1863
-    valuesByField[fieldName] = value;
+
+    valuesByField![field] = value;
 }
 
-export function getHiddenField(o: any, fieldName: symbol) {
+export function getHiddenField(o: any, field: HiddenField): unknown {
```

Comment:
should the return be unknown or any?

Reply:
I think that `unknown` is safer in this case because it forces the caller to check that is the returned value via an assertion, or using a typescript type guard. While using `any` would silently fail if there is a type mismatch.